packages/app/src/cli/models/app/app.ts

```diff
@@ -322,7 +322,7 @@ function findExtensionByHandle(allExtensions: ExtensionInstance[], handle: strin
 }
 
```

Comment:
I removed the previous `webhooksConfiguration(configuration: object) ` and ` simplifySubscriptions(subscriptions: WebhookSubscription[] | undefined)` functions b/c of this PR which is related to the `link` command - @alvaro-shopify does this seem correct?

Reply:
For context, here's what the old code block looked like:

```

function webhooksConfiguration(configuration: object) {
  const webhooks = getPathValue<WebhooksConfig>(configuration, 'webhooks') as WebhooksConfig
  if (webhooks?.subscriptions) {
    const simplifiedSubscritions = simplifySubscriptions(webhooks.subscriptions)
    setPathValue(webhooks, 'subscriptions', simplifiedSubscritions)
  }
  return {webhooks}
}

function simplifySubscriptions(subscriptions: WebhookSubscription[] | undefined): WebhookSubscription[] {
  if (!subscriptions) return []

  return subscriptions.reduce((accumulator, subscription) => {
    const existingSubscription = accumulator.find(
      (sub) =>
        sub.uri === subscription.uri &&
        sub.sub_topic === subscription.sub_topic &&
        sub.include_fields === subscription.include_fields &&
        sub.metafield_namespaces === subscription.metafield_namespaces,
    )
    if (existingSubscription) {
      if (subscription.compliance_topics) {
        existingSubscription.compliance_topics ??= []
        existingSubscription.compliance_topics.push(...subscription.compliance_topics)
      }
      if (subscription.topics) {
        existingSubscription.topics ??= []
        existingSubscription.topics.push(...subscription.topics)
      }
    } else {
      accumulator.push(subscription)
    }
    return accumulator
  }, [] as WebhookSubscription[])
}
```

But all the different configuration paths has now been removed (including the `webhooksConfiguration`) so I am assuming we don't need this any more?

Reply:
Right, that simplify logic is still required, but we have to find a new place. The purpose is that, when you have a topic and a compliance topic sharing the same URL, both should be merged in the TOML under the same subscription:
```
  [[webhooks.subscriptions]]
  uri = "https://example.com/webhooks"
  topics = ["products/update"]
  compliance_topics = [ "customers/data_request" ]
``` 
Álvaro suggested a couple of alternatives:
- Restore the removed logic, but fixing the issue that [this PR](https://github.com/Shopify/cli/pull/3540) solved in a different way
- Add a `simplify` function to the specs and call each one after building the full object

I think the second way can be simpler.